presto-docs/src/main/sphinx/functions/lambda.rst

```diff
@@ -60,6 +60,16 @@ Lambda Functions
                       (s, x) -> CAST(ROW(x + s.sum, s.count + 1) AS ROW(sum DOUBLE, count INTEGER)),
                       s -> IF(s.count = 0, NULL, s.sum / s.count));
 
+.. function:: regexp_replace(string, pattern, function) -> varchar
+
+    Replaces every instance of the substring matched by the regular expression
+    ``pattern`` in ``string`` using ``function``. `Capturing groups`_ are passed
+    to ``function`` as an array, starting from the second capturing group for
```

Comment:
Have you decided on not passing group 0 (entire match)? Wouldn't that be useful too (I know that there is problem with array indexes counting from 1)? What if we pass map instead of an array, e.g:
```
 map<integer, varchar>
```
then we would pass group 0 too, right? e.g:
```
groups -> groups[0] || group[1] || .. || group[n]
```
Such convention might be slower, but seems more convenient.

Reply:
The reason I skipped group 0 was to align with the numbering in current regexp_replace implementations. In this Presto document example, the matched group $2 is the match for ([ab]). 
SELECT regexp_replace('1a 2b 14m', '(\d+)([ab]) ', '3c$2 '); -- '3ca 3cb 14m'

The users can always add a () to the pattern to get the entire match.

Reply:
Another reason why we don't want to pass the whole match is the performance implications. Always passing the whole match will incur additional cost, and it's not always used by the user. Since there is workaround (by adding ()) for the user to get the entire match, we can just skip the entire match.